ocaml/lambda/simplif.ml

```diff
@@ -156,18 +156,24 @@ let simplify_exits lam =
   | Lstaticraise (i,ls) ->
       incr_exit i 1 try_depth;
       List.iter (count ~try_depth) ls
-  | Lstaticcatch (l1,(i,[]),Lstaticraise (j,[]), _) ->
+  | Lstaticcatch (l1,(i,[]),Lstaticraise (j,[]), _, _) ->
```

Comment:
It seems to me this can make a pop_region disappear?

Reply:
Not as long as the original lambda was well-formed. If this `Lstaticcatch` is in the parent region then it can only jump to other handlers that are also in the parent region.

The key point here is that `pop_region` is an annotation describing what region the handler is in, not an instruction to insert an `End_region` construct. As said above, I think perhaps a rename would make this clearer.